docs/examples/use_cases/pytorch/efficientnet/image_classification/dataloaders.py

```diff
@@ -274,6 +274,190 @@ def expand(num_classes, dtype, tensor):
+        )
+
+    transforms_list.append(transforms.ToTensor())
+    transforms_list.append(
+        transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+    )
+    train_dataset = datasets.ImageFolder(
+        traindir, transforms.Compose(transforms_list)
+    )
+
+    if torch.distributed.is_initialized():
+        train_sampler = torch.utils.data.distributed.DistributedSampler(
+            train_dataset, shuffle=True
+        )
+    else:
+        train_sampler = None
+
+    train_loader = torch.utils.data.DataLoader(
+        train_dataset,
+        sampler=train_sampler,
+        batch_size=batch_size,
+        shuffle=(train_sampler is None),
+        num_workers=workers,
+        worker_init_fn=_worker_init_fn,
+        pin_memory=True,
+        collate_fn=partial(fast_collate, memory_format),
+        drop_last=True,
+        persistent_workers=True,
+        prefetch_factor=prefetch_factor,
+    )
+
+    return (
+        PrefetchedWrapper(
+            train_loader,
+            start_epoch,
+            num_classes,
+            one_hot,
+        ),
+        len(train_loader),
+    )
+
+
+def get_pytorch_val_loader(
+    data_path,
+    image_size,
+    batch_size,
+    num_classes,
+    one_hot,
+    interpolation="bilinear",
+    workers=5,
+    _worker_init_fn=None,
+    crop_padding=32,
+    memory_format=torch.contiguous_format,
+    prefetch_factor=2,
+):
+    interpolation = {"bicubic": Image.BICUBIC, "bilinear": Image.BILINEAR}[
+        interpolation
+    ]
+    valdir = os.path.join(data_path, "val")
+    transforms_list = [
+        transforms.Resize(
+            image_size + crop_padding, interpolation=interpolation
+        ),
+        transforms.CenterCrop(image_size),
+    ]
+    transforms_list.append(transforms.ToTensor())
+    transforms_list.append(
+        transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+    )
+    val_dataset = datasets.ImageFolder(
+        valdir,
+        transforms.Compose(transforms_list),
+    )
+
+    if torch.distributed.is_initialized():
+        val_sampler = torch.utils.data.distributed.DistributedSampler(
+            val_dataset, shuffle=False
+        )
+    else:
+        val_sampler = None
+
+    val_loader = torch.utils.data.DataLoader(
+        val_dataset,
+        sampler=val_sampler,
+        batch_size=batch_size,
+        shuffle=(val_sampler is None),
+        num_workers=workers,
+        worker_init_fn=_worker_init_fn,
+        pin_memory=True,
+        collate_fn=partial(fast_collate, memory_format),
+        drop_last=False,
+        persistent_workers=True,
+        prefetch_factor=prefetch_factor,
+    )
+
+    return PrefetchedWrapper(val_loader, 0, num_classes, one_hot), len(
+        val_loader
+    )
+
+
+def fast_optimized_collate(memory_format, batch):
+    imgs = [img[0] for img in batch]
+    targets = torch.tensor([target[1] for target in batch], dtype=torch.int64)
+    w = imgs[0].size[0]
+    h = imgs[0].size[1]
```

Comment:
I guess it's OK here since it's a PIL image and not a tensor - a comment would be nice.
```suggestion
    # PIL.Image.size returns (width, height)
    w, h = imgs[0].size
```

Reply:
Fixed